src/libcrun/linux.c

```diff
@@ -2036,8 +2037,14 @@ uidgidmap_helper (char *helper, pid_t pid, char *map_file, libcrun_error_t *err)
       args[nargs++] = p;
     }
   args[nargs++] = NULL;
+  ret = run_process(args, err);
+  if (UNLIKELY(ret < 0))
+    {
+      libcrun_warning ("uidgidmap: unable to invoke %s, will try creating a user namespace as an alternative", helper);
```

Comment:
I think we should move this warning higher in the call stack and verify that `def->linux->uid_mappings_len ==0` or `def->linux->gid_mappings_len ==0` because if they are set, failing with uidgidmap is already a hard error

Reply:
Sure that makes sense. I'll do that instead

Reply:
@giuseppe Force pushed with the corrections, could you take a look.